crates/cargo-contract/src/main.rs

```diff
@@ -486,14 +486,21 @@ fn main() {
     tracing_subscriber::fmt::init();
 
     let Opts::Contract(args) = Opts::parse();
+    let suppress_err = match &args.cmd {
+        Command::Call(call_cmd) if call_cmd.is_json() => true,
+        Command::Instantiate(inst_cmd) if inst_cmd.is_json() => true,
+        _ => false,
+    };
     match exec(args.cmd) {
         Ok(()) => {}
         Err(err) => {
-            eprintln!(
-                "{} {}",
-                "ERROR:".bright_red().bold(),
-                format!("{:?}", err).bright_red()
-            );
+            if !suppress_err {
```

Comment:
I think we actually want to report the error as json rather than supressing it. At the moment for some errors we will get the pretty printed json at the call site, and all others will not be displayed at all.

Reply:
If don't do this, then the output result can be like this:
```json
{
  "module_error": {
    "pallet": "Contracts",
    "error": "DuplicateContract",
    "docs": [
      "A contract with the same AccountId already exists."
    ]
  }
}
{
  "error": "Pre-submission dry-run failed. Use --skip-dry-run to skip this step."
}
```

Reply:
Then we will need to rethink the approach. How about instead of printing the json error directly, we bubble it back up as an error?.

So instead of `eprintnln!("{}", json_error); Ok(())` you could do `(Err(anyhow!("{}", json_err))`. Then this would be printed correctly at the top level

Reply:
I made the error propagate to the top level, all the way back to the `main()`, then I check if an error is an json format and print out, otherwise I wrap the message into a serializable object.